src/mpi/comm/commutil.c

```diff
@@ -904,14 +904,14 @@ int MPII_Comm_copy(MPIR_Comm * comm_ptr, int size, MPIR_Info * info, MPIR_Comm *
     }
     MPID_THREAD_CS_EXIT(POBJ, comm_ptr->mutex);
 
-    if (info) {
-        MPII_Comm_set_hints(newcomm_ptr, info);
-    }
-
     newcomm_ptr->tainted = comm_ptr->tainted;
     mpi_errno = MPIR_Comm_commit(newcomm_ptr);
     MPIR_ERR_CHECK(mpi_errno);
 
+    if (info) {
+        MPII_Comm_set_hints(newcomm_ptr, info);
+    }
+
```

Comment:
This is wrong. In general, `Comm_commit` need know the hints. Thus we need set hints before  commit. If device is not ready to handle the hints before commit, and then don't do anything in `MPID_Comm_set_hints`. Do it in `MPID_Comm_commit` instead.

Reply:
Handled the problem differently